components/org.wso2.carbon.identity.application.authenticator.samlsso/src/test/java/org/wso2/carbon/identity/application/authenticator/samlsso/util/SSOUtilsTest.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.application.authenticator.samlsso.util;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.application.authenticator.samlsso.TestConstants;
+import org.wso2.carbon.identity.application.authenticator.samlsso.exception.SAMLSSOException;
+import org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Unit test cases for SSOUtils
+ */
+public class SSOUtilsTest {
+
+    @Test
+    public void testCreateID() {
+
+        Assert.assertNotNull(SSOUtils.createID(), "Failed to generate an ID");
+    }
+
+    @Test
+    public void testEncode() {
+
+        Assert.assertEquals(SSOUtils.encode(TestConstants.XML_STRING), TestConstants.ENCODED_STRING);
+    }
+
+    @Test
+    public void testDecode() throws SAMLSSOException {
+
+        Assert.assertEquals(SSOUtils.decode(TestConstants.REDIRECT_REQUEST), TestConstants.REDIRECT_DECODED_STRING);
+    }
+
+    @Test (expectedExceptions = RuntimeException.class)
+    public void testDecodeEmptyRequest() throws SAMLSSOException {
+
+        String emptyRedirectRequest = "";
+        SSOUtils.decode(emptyRedirectRequest);
+    }
+
+    @Test(expectedExceptions = {SAMLSSOException.class})
+    public void testDecodeForInvalidRequest() throws SAMLSSOException {
+
+        SSOUtils.decode(TestConstants.INVALID_REDIRECT_REQUEST);
+    }
+
+    @Test
+    public void testDecodeForPost() throws SAMLSSOException {
+
+        Assert.assertEquals(SSOUtils.decodeForPost(TestConstants.POST_REQUEST), TestConstants.POST_DECODED_STRING);
+    }
+
+    @Test
+    public void testIsAuthnRequestSigned() {
+
+        Assert.assertEquals(false, SSOUtils.isAuthnRequestSigned(null));
+
+        Map<String, String> properties = new HashMap<>();
+        Assert.assertEquals(false, SSOUtils.isAuthnRequestSigned(properties));
```

Comment:
we can use asserFalse instead of assertEquals here.

Reply:
+1